### PR TITLE
feat: fetch latest nonce from chain state when initialising the BlockchainService

### PR DESCRIPTION
**Pull request by a-moreira**
Solves the issue of restarting the nonce after node initialisation.

### CODE REVIEW

**Review by links234 — APPROVED** [review]
Very nice!

File: node/src/services/blockchain/handler.rs [hunk 1]
```diff
@@ -272,12 +274,21 @@ impl BlockchainService {
         rpc_handlers: Arc<RpcHandlers>,
         keystore: KeystorePtr,
     ) -> Self {
+        // Get best nonce (not pending) for one of the pub keys (in this case the last one)
+        // in our vector of keys in the Keystore.
+        let current_block_hash = client.chain_info().best_hash;
```
Comment (ffarall) [inline node/src/services/blockchain/handler.rs:279]: Although better than before, I believe this won't solve our problems definitively. This code executes before the node syncs the chain with its peers. Meaning that if there was a transaction for this `pub_key` after this node's last block recorded (which when initialising from scratch is 0), this won't return the correct nonce.

**Review by ffarall — CHANGES_REQUESTED** [review]

File: node/src/services/blockchain/handler.rs (hunk 1 above)
Comment (a-moreira) [reply node/src/services/blockchain/handler.rs]: This is a good point. I changed it so we start with 0 and update the nonce after each new block notification is identified by the service.

**Review by ffarall — APPROVED** [review]
Great job!

**Pull request merged**